### ISSUE TITLE
Add background color to social icons in default theme appearance

### ISSUE THREAD

**Issue by germainelee** [owncast/owncast#3747]
### Edit 2: Acceptance criteria-
- Add `background-color` using the default theme's component color var `--theme-color-components-content-background`
- Set `border-radius: 20px` or use variable if there is one so that it appears circular
- Add `padding` if needed

##  **tldr;** it would look something like this 
 - on light background, background colour should look nearly invisible 
  ![Image](https://github.com/user-attachments/assets/57bba408-c400-4742-82a2-2217295cb56b)

- on dark background, background colour should look visible 
 ![Image](https://github.com/user-attachments/assets/476873cf-2c2e-46fe-96a3-32a362c29b00)
<hr>
OMG! I love that there's dark mode! Good job!

Although the colors of the Autheticate modal are working well in light mode, it seems the accordion background color in IndieAuth and FediAuth tabs didn't have a dark mode version causing readability issues with white text on white baground :(

![Screenshot_20240528_235319_Chrome](https://github.com/owncast/owncast/assets/118464031/9a124ba3-fded-4287-9cfc-1eacf9668ec5)

My workaround to read the text was to highlight / select the words.

**Comment by germainelee** [comment]
And the color of the horizontal line for the Tabs needs dark mode version too.

Edit: Not sure if this occurs due to the server's customisation.

**Comment by gabek** [comment]
Unfortunately, I'm not sure what you're seeing, since there is no dark mode in Owncast, there's just the single color scheme. Could you share how you're getting to see those colors?

**Comment by germainelee** [comment]
This issue somehow appears in that channel only... as dark mode is not applied to others such as [this channel](https://owncast.funcube.space/)

Steps to replicate: 

1. visit [https://radio.gamethattune.com/](https://radio.gamethattune.com/)
2. Make your OS settings to be dark mode
![image](https://github.com/owncast/owncast/assets/118464031/dbc77962-5ba9-46e4-adb3-6aabb7adb88f)

3. Click on username
4. Choose Authenticate
5. Observe

<blockquote><img src="https://radio.gamethattune.com/thumbnail.jpg" width="48" align="right"><div><img src="/img/favicon/favicon-16x16.png" height="14"> Game That Tune Radio</div><div><strong><a href="https://radio.gamethattune.com/">Game That Tune Radio</a></strong></div><div>The never-ending video game music livestream featuring guessing games and always taking your requests!

#vgm #music #video games #radio #gamemusic #owncast #fedi22</div></blockquote>

**Comment by gabek** [comment]
They might have their own custom styling that supports multiple color schemes.

**Comment by germainelee** [comment]
That's pretty cool! Thank for clarification!

If I understood you correctly, the button icons below the tags are barely visible due to the custom styling, and to ensure sufficient contrast, they need to style the background accordingly. So, I'm thinking that it would be nice if the logos have a light and dark version.

![image](https://github.com/owncast/owncast/assets/118464031/60f9cfd0-08d3-46a9-a233-ab59ca8ade1c)

Logos/icons blends with dark background...
![image](https://github.com/owncast/owncast/assets/118464031/6cd6af69-a952-4b2c-b517-b90ac0dc2624)

unless the admin adds a Light background.
![image](https://github.com/owncast/owncast/assets/118464031/37e0daa6-0f76-4c28-87f0-e8034c321e73)

**Comment by gabek** [comment]
@germainelee Yeah, since Owncast only has one color scheme, any changes that somebody makes it's up to them to design it in a way that makes sense. There would be no way for Owncast to swap the icons out automatically, it has no idea what the page looks like. The Owncast demo server's icons look bad, simply because I'm bad at making things look good. So that's my fault personally.

**Comment by germainelee** [comment]
@gabek If that's the case, how about we give them the option to customise the button style?

If they choose white, all the logo/icon will be for light theme
![image](https://github.com/owncast/owncast/assets/118464031/3ef84320-a5d7-4cfa-a56c-94d49205cd45)
![image](https://github.com/owncast/owncast/assets/118464031/9909c2ad-a5be-46c3-888e-25f526d69010)

If they choose black, all the logo/icon will be for dark theme
![image](https://github.com/owncast/owncast/assets/118464031/2ef81ada-0fb4-40fc-b475-62af2e13c1d2)
![image](https://github.com/owncast/owncast/assets/118464031/03a6f779-7103-424e-9df6-e2961134bebe)

**Comment by gabek** [comment]
It's not a bad idea, but we do allow people to change their appearance in any way they like, including their button style. So I don't think we need to have overly specific appearance settings for this when it's already 100% customizable.

**Comment by germainelee** [comment]
Enhancement maybe?

**Comment by gabek** [comment]
The default theme of Owncast looks fine. So to change that theme, somebody has to manually create a new one. If they create one that doesn't look fine, I don't think it's up to Owncast to work around their design and implementation decisions. If they want icons to look differently, then they can change them to look any way they wish.

If we had an actual built-in dark and light theme, then having a switch between them would make sense. But we only have one color scheme and people customize it.

**Comment by germainelee** [comment]
I'm not referring to owncast light-dark theming. it's specifically to give them an option to choose icon on either light bg or icon on dark bg because on dark bg, the default web icon has poor contrast and I believe majority of Admins don't bother to add a background to make it visible for reasons.
![image](https://github.com/user-attachments/assets/adb6ec79-12d1-4d8c-b029-178f5743e19e)

**Comment by gabek** [comment]
I meant if we did have a dark theme, then it would make sense to change the colors of icons to reflect it.

> I believe majority of Admins don't bother to add a background to make it visible for reasons.

That's the thing. If people are customizing their appearance, then they're responsible for the appearance. Just like how I made the demo server look bad, that's my fault. I should spend more time making it look good. That's the responsibility of somebody who's customizing their page. We can't add a toggle for everything on the page to support the cases that somebody made a bad decision.

As long as our default theme looks good, then it's up to each individual for the rest.

**Comment by germainelee** [comment]
I'm thinking the default theme could help them a little by setting a `background-color` of low opacity so that on light theme, it looks nearly invisible and looks visible on dark. If they don't like the style, they could tweak it. See images below for on how it looks on light and dark backgrounds:
![Image](https://github.com/user-attachments/assets/c2d8a15f-6cb3-4d2f-b3b2-40320468baf0)
![Image](https://github.com/user-attachments/assets/59d94f62-a016-4c6c-a8ec-f0d19d76a697)

**Comment by gabek** [comment]
> I'm thinking the default theme could help them a little by setting a background-color of low opacity so that on light theme, it looks nearly invisible and looks visible on dark. 

If we can tweak the default theme and improve things for everyone, without making the default look bad, I'm totally on board!

**Comment by germainelee** [comment]
tweaking the default theme seems like it requires lesser dev effort compared to adding a new option with separate set of icons on light background and on dark background . Added acceptance criteria in 1st comment, https://github.com/owncast/owncast/issues/3747#issue-2322310807 for anyone who's interested in picking this up

**Comment by germainelee** [comment]
not stale. 

is this a good first issue?
